INT-5174 Technical Documentation Improvements (#131)
sonatype/helm3-charts: overview.md

    distribution and is available at https://www.eclipse.org/legal/epl-2.0/.

-->

# Document Purpose

This page provides high-level technical information regarding the **Helm3 Charts for Sonatype Products**

## Product Overview

[Helm](https://helm.sh/) is a package manager and templating engine for installing software in [Kubernetes](https://kubernetes.io/). We provide our [own Helm Charts](https://sonatype.github.io/helm3-charts/) to help people install our servers in Kubernetes. 

These charts are indexed and promoted at [ArtifactHub](https://artifacthub.io/packages/search?page=1&repo=sonatype). The index we publish lives in a GitHub Pages site: https://sonatype.github.io/helm3-charts/index.yaml. With helm charts, the user can change settings by modifying a copy of the `values.yaml` before installing the chart.

## High-Level Technical Description

As described earlier in this project we use Helm, a package manager for Kubernetes. Helm will help us to quickly install/upgrade/maintain Nexus IQ and Nexus Repository Manager on a Kubernetes cluster.

In this repository, we have two charts: 
* Nexus IQ `\charts\nexus-iq`

> * Nexus Repository Manager `\charts\nexus-repository-manager`

Reviewer: should these be forward slashes instead of windows backslashes?

Author: Yeahp that is a good point, I am going to change them

<!--

    Copyright (c) 2019-present Sonatype, Inc.
    This program and the accompanying materials are made available under
    the terms of the Eclipse Public License 2.0 which accompanies this
    distribution and is available at https://www.eclipse.org/legal/epl-2.0/.

-->

# Document Purpose

This page provides high-level technical information regarding the **Helm3 Charts for Sonatype Products**

## Product Overview

> [Helm](https://helm.sh/) is a package manager and templating engine for installing softare in [Kubernetes](https://kubernetes.io/). We provide our [own Helm Charts](https://sonatype.github.io/helm3-charts/) to help people install our servers in Kubernetes. 

Reviewer: typo: software

Author: oops, fixed!

> These charts are indexed and promoted at ArtifactHub. The index we publish lives in a GitHub Pages site: https://sonatype.github.io/helm3-charts/index.yaml. With helm charts, the user can change settings by modifying a copy of the `values.yaml` before installing the chart.

Reviewer: may as well link to artifact hub.

Author: Makes sense, I just added the link

## High-Level Technical Description

> As described earlier in this project we use Helm, a package manager for Kubernetes. Helm will help us to quickly install/upgrade/mantain Nexus IQ and Nexus Repository Manager on a Kubernetes cluster.

Reviewer: typo: maintain

In the repository we have two Charts, one for Nexus IQ `\charts\nexus-iq`,  and other for Nexus Repository Manager `\charts\nexus-repository-manager`. 

> Each Charts is composed for a set of **templates** that are basically `yaml` files that describe the different Kubernetes resources along with a templating syntax to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different variables that you will find in the different **templates**.

Reviewer 2: Is this supposed be spelled this way?
```suggestion
Each chart is composed for a set of **templates** that are basically `yaml` files that describe the different Kubernetes resources along with a templating syntax to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different variables that you will find in the different **templates**.
```

> Each Charts is composed for a set of **templates** that are basically `yaml` files that describe the different Kubernetes resources along with a templating syntax to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different variables that you will find in the different **templates**.

Reviewer 2: ```suggestion
Each Charts is composed for a set of **templates** that are `yaml` files that describe the different Kubernetes resources along with a templating syntax to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different variables that you will find in the different **templates**.
```

> Each Charts is composed for a set of **templates** that are basically `yaml` files that descibe the different Kubernetes resources along with a templating syntaxys to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different varaibles that you will find in the different **templates**.

Reviewer 3: ```suggestion
Each Charts is composed for a set of **templates** that are basically `yaml` files that describe the different Kubernetes resources along with a templating syntax to define variables, that is used by Helm. Also, for each chart, we have a `values.yaml` file, with the default values for the different variables that you will find in the different **templates**.
```

Finally these charts are indexed and promoted at ArtifactHub. The Helm repository's `index.yaml` and all charts binaries are in `\docs` folder. The index we publish lives in a **GitHub Pages site**: https://sonatype.github.io/helm3-charts/index.yaml. The repository is exposed to the public using **GitHub pages**.

## Internal Dependencies

Here are some internal dependencies

- Nexus Repository Manager Docker Image
- Nexus IQ Docker Image

## External Dependencies

Here are some external dependencies

- GitHub pages
- Artifact Hub

## Data Persistence 
N/A

## Localhost Development
See [Helm Charts](https://docs.sonatype.com/display/INT/Helm+Charts) internal documentation.

## Architecture Diagram
See [Helm Charts](https://docs.sonatype.com/display/INT/Helm+Charts) internal documentation.

## Release Plan

> See [Helm Charts](https://docs.sonatype.com/display/INT/Helm+Charts) internal documentation.

Reviewer 2: This is a public project and these are internal wiki links. What is our protocol here?

Author: You are right, that is a good point. Do I need to create a new page on a public site?

Reviewer: the readme in the project already has some of this info, so maybe just build out a little more in there and reference that?

Reviewer 2: Note that the contributing doc in this repo is not a link to our wiki like our other projects, but rather to something in `docs/`.
I would vote that we:
1. Drop the 'Localhost Development' and the readme.md should have anything relevant to doing this development (I assume it already does)
2. Drop 'Arch Diagram'... not really important
3. Drop 'Release Plan'. Is the release process something always internal? i.e. its in our wiki already and we know its there so I would say just drop this. If it is something the community needs to know it should be in the readme?

Author: Ok, then I am going to update the README with the needed information and reference from here. 

I will remove Release Plan and Architecture Diagram sections. I'll keep the Local Development section and point it out to the README file.

Author: Regarding the contributing doc, should I take any action? Like adding the link to our confluence page?

Author: I was looking in more detail. It looks like the `CONTRIBUTING.md` has some information related to signing a Sonatye CLA and mentions a gitter channel.  Is that information relevant/updated? 

Maybe, should I update the `README.md` to point to our confluence page, and delete the `CONTRIBUTING.md` file?

Reviewer: To @collinpeters' point, confluence (docs.s.c) links will be unavailable to the general public who could be reading this public project, so it's probably going to look a little different from some of our other private projects. We probably want to keep as much info as possible available in MD files of this repo, without linking/divulging internal info, because that's just frustrating and inaccessible to the public. Externally, we can mention that sonatype does the releases, and the internal docs can describe that process and link to the jobs.

Reviewer 2: No. This is a public project and those are the correct details for someone to contribute a PR. The gitter channel is monitored by the DevEx team AFAIK (CC @bhamail, is that still true).
In any case, this is a rarer case of a project supported by Integrations, but not under community org. My greater point is that it seems silly to have links in a public project to our internal wiki that outside folks can't access.

Reviewer: contributing.md doc should stay, since that directs outsiders on how to submit PRs to this public project.

Author: Ok, perfect. I just updated the `overview.md` to remove any reference to internal confluence pages. I also updated the `README.md` to add the steps to test locally. @jflinchbaugh do you think we should add something else to the README.md?
